### ISSUE TITLE
Getting error from estAvgError.R (ABBA-BABA)!!!

### ISSUE THREAD

**Issue by Razi1026** [ANGSD/angsd#219]
Hi all,

I have an urgent problem. I need to run the estAvgError.R script. This is the command that I used:

Rscript ./estAvgError.R angsdFile="sylvia" out="result" sizeFile="sizeFile.size" nameFile="popNames.name"

But I get this error:
Error in seq.default(1, lenList, numComb) : invalid '(to - from)/by'

Could you please help me to pass this error?

Regards,
Razi

**Comment by Razi1026** [comment]
Hi again, 

I could solve this problem. I think it was because I used 3 populations for abba-baba! but this test is for 4 populations.
I could get some result files and tables. But I got another error: 
Building tree error matrices: Error in outDataTotal[idxList, ] : subscript out of bounds
Execution halted
Would you please help me?

Thanks
Razi

**Comment by SamueleSoraggi** [comment]
Hi Razi,

sorry for the late reply;
it sounds like the first error was indeed happening because of not using
enough populations.
Would you double check that in the size file and name file you put the
correct informations in? Just write here the content of the two files so I
can see it :)

Cheers

Den søn. 16. jun. 2019 kl. 14.14 skrev Raziyeh <notifications@github.com>:

**Comment by Razi1026** [comment]
Hi,

**sizefile.size** 
5
3
2
1

**popNames.name**
H1
H1
H1
H1
H1
H2
H2
H2
H3
H3
O

**bam.filelist**

P11456_101.RG.dup.sorted.bam
P11456_102.RG.dup.sorted.bam
P11456_105.RG.dup.sorted.bam
P11456_106.RG.dup.sorted.bam
P10261_104.RG.dup.sorted.bam
P11456_103.RG.dup.sorted.bam
P11456_104.RG.dup.sorted.bam
P4253_106.RG.dup.sorted.bam
P11081_121.RG.dup.sorted.bam
P11081_122.RG.dup.sorted.bam
P11081_126_100bp.SDRG.bam


**my first script**

angsd -b bam.filelist -ref SylviaAllPath.scf.fasta -remove_bads 1 -uniqueOnly 1 -only_proper_pairs 1 -C 50 -baq 1 -minMapQ 30 -minQ 25 -minInd 2.25 -minIndDepth 3 -doCounts 1 -setMinDepth 9 -setMaxDepth 300 -doAbbababa2 1 -blockSize 1000000 -maxDepth 2400 -sizeFile sizefile.size -P 10 -out sylvia -useLast 1

**The second script**

Rscript ./estAvgError.R angsdFile="sylvia" out="result" sizeFile="sizefile.size" nameFile="popNames.name" &>  Dstat_table.txt

***As you see, I have 3 species and 1 outgroup. But I don't know should I count outgroup as a species when I want to calculate -minInd, setMinDepth and setMaxDepth? here, I didn't do.


Thanks for your help.

Best

**Comment by SamueleSoraggi** [comment]
Dear Razi,

the population name file should contain only the name of each population.
Change the file content into
H1
H2
H3
O
and let me know if it works. The rest seems fine at a first look.

Cheers

Den man. 17. jun. 2019 kl. 11.40 skrev Raziyeh <notifications@github.com>:

**Comment by Razi1026** [comment]
Ok, Thanks.

I changed population names. This time, there is not this error: Error in outDataTotal[idxList, ] : subscript out of bounds Execution halted.


But, again it say: Building tree error matrices...So I don't have any tree!

**Comment by Razi1026** [comment]
![Capture](https://user-images.githubusercontent.com/45821726/59658068-e25d3700-91a2-11e9-8463-41a829d9260e.JPG)

**Comment by SamueleSoraggi** [comment]
Hi Riza,

the message "building tree error matrices" means that the software is using
estimated rates of transitions/transversions in case they are provided.
This is not happening in your case, only the message is shown. Then you can
see that you are getting values of the D-statistic in output for different
combinations of your 4 populations, that is, considering the possible trees
having fixed outgroup O:
```
( ( ( H1, H2 ) H3 ) O )
( ( ( H1, H3 ) H2 ) O )
( ( ( H2, H3 ) H1 ) O )
```
You can find the results also in the output file text that has prefix
`result` as you specified when you ran the command.
In other words, everything is working as expected.

Cheers

Den tir. 18. jun. 2019 kl. 08.27 skrev Raziyeh <notifications@github.com>:

**Comment by Razi1026** [comment]
Hi Samuele,

Thanks for your comment. So, where is the tree? I have just 2 result files: result.Observed.txt and result.TransRem.txt and there is no tree!


again thanks for your help.

Best

**Comment by SamueleSoraggi** [comment]
Hi Razi,

`result.Observed.txt` gives you the D-statistic for the possible trees
having O as a fixed outgroup. The order of the populations in the tree is
illustrated in the file, and you are interested in the combination of
populations H1, H2, H3, O.
The software does not output 'a tree' but just the result of a statistical
test (the D-statistic test) based on a hypothesized tree involving H1, H2,
H3, O in absence of admixture.

The same results are in the file `result.TransRem.txt`, with the only
difference that ancient transitions are not considered when calculating the
D-statistic (useful for ancient datasets).

Cheers

Den tir. 18. jun. 2019 kl. 09.32 skrev Raziyeh <notifications@github.com>:

**Comment by Razi1026** [comment]
Thanks for your advice :)
